contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-rule.y

```diff
@@ -1640,33 +1643,34 @@ tsql_pivot_expr: TSQL_PIVOT '(' func_application FOR columnref IN_P in_expr ')'
 					a_star_restarget->indirection = NIL;
 					a_star_restarget->val = (Node *) a_star;
 					a_star_restarget->location = -1;
-					
 
 					/* prepare aggregation function for pivot source sql */
 					restarget_aggfunc->name = NULL;
 					restarget_aggfunc->name_location = -1;
 					restarget_aggfunc->indirection = NIL;
 					restarget_aggfunc->val = (Node *) $3;
 					restarget_aggfunc->location = -1;
-					
-					if (IsA((List *)$7, List))
+
+					column_list = (List *)$8;
+					column_const_list = NIL;
+					value_col = list_nth_node(ColumnRef, ((FuncCall *) $3)->args, 0);
+					value_colstr = list_nth_node(String, value_col->fields, ((List *)value_col->fields)->length - 1);
+
+					foreach(lc ,column_list)
 					{
-						for (int i = 0; i < ((List *)$7)->length; i++)
-						{
-							ColumnRef	*tempRef = list_nth((List *)$7, i);
-							String		*s = list_nth(tempRef->fields, 0);
-							Node		*n = makeStringConst(s->sval, -1);
-							List 		*l = list_make1(copyObject(n));
-							if (value_col_strlist == NULL || subsel_valuelists == NULL)
-							{
-								value_col_strlist = list_make1(s);
-								subsel_valuelists = list_make1(l);
-							}else
-							{
-								value_col_strlist = lappend(value_col_strlist, s);
-								subsel_valuelists = lappend(subsel_valuelists, l);
-							}
-						}
+						Node 	*conlumn_const;
+						String 	*column_str;
+
+						column_str = (String *)lfirst(lc);
+
+						if (column_str != NULL && strcmp(column_str->sval, value_colstr->sval) == 0)
+							ereport(ERROR,
+									(errcode(ERRCODE_SYNTAX_ERROR),
+										errmsg("The column name \"%s\" specified in the PIVOT operator conflicts with the existing column name in the PIVOT argument.", value_colstr->sval),
```

Comment:
Is this the same error msg as SQL Server?

Reply:
Yes, same error msg as SQL Server